ipamd/ipamd.go

```diff
@@ -582,44 +579,36 @@ func (c *IPAMContext) increaseIPPool() {
 		return
 	}
 
-	if c.maxENI == c.dataStore.GetENIs() {
-		log.Debugf("Skipping increase IP pool due to max ENI already attached to the instance: %d", c.maxENI)
-		return
-	}
-
-	// Unknown instance type and MAX_ENI is not set
-	if c.maxENI == 0 {
-		log.Errorf("Unknown instance type and MAX_ENI is not set.  Cannot increase IP pool.")
-	}
-
-	if c.dataStore.GetENIs() < c.maxENI {
-		c.tryAllocateENI()
-		c.updateLastNodeIPPoolAction()
-	} else {
-		log.Debugf("Skipping ENI allocation due to max ENI already attached to the instance: %d", c.maxENI)
-	}
-
+	// Try to add more IPs to existing ENIs first.
 	increasedPool, err := c.tryAssignIPs()
 	if err != nil {
 		log.Errorf(err.Error())
```

Comment:
What about `ipamdErrInc`?

Reply:
I only increase it after both trying all and then 1 ip:

https://github.com/aws/amazon-vpc-cni-k8s/pull/487/files#diff-3fbbb0bd392fd47bd85a56d6221de538R682